server/models/models.go

```diff
@@ -261,9 +263,18 @@ type UpdateSubscription struct {
 	SourceID   string `json:"source_id,omitempty"`
 	EndpointID string `json:"endpoint_id,omitempty"`
 
-	AlertConfig  *datastore.AlertConfiguration  `json:"alert_config,omitempty"`
-	RetryConfig  *datastore.RetryConfiguration  `json:"retry_config,omitempty"`
-	FilterConfig *datastore.FilterConfiguration `json:"filter_config,omitempty"`
+	AlertConfig     *datastore.AlertConfiguration     `json:"alert_config,omitempty"`
+	RetryConfig     *RetryConfiguration               `json:"retry_config,omitempty"`
+	FilterConfig    *datastore.FilterConfiguration    `json:"filter_config,omitempty"`
+	RateLimitConfig *datastore.RateLimitConfiguration `json:"rate_limit_config,omitempty"`
+	DisableEndpoint *bool                             `json:"disable_endpoint" bson:"disable_endpoint"`
+}
+
+type RetryConfiguration struct {
+	Type            datastore.StrategyProvider `json:"type,omitempty" valid:"supported_retry_strategy~please provide a valid retry strategy type"`
+	Duration        string                     `json:"duration,omitempty" valid:"duration~please provide a valid time duration"`
+	IntervalSeconds uint64                     `json:"interval_seconds" valid:"int~please provide a valid interval seconds"`
+	RetryCount      uint64                     `json:"retry_count" valid:"int~please provide a valid retry count"`
```

Comment:
Why is this being defined here, I thought there it was already defined in the models package?

Reply:
@jirevwe We can resolve this comment yeah?